inference-engine/tests/unit/gna/ngraph/transformations/handle_transposes_around_matmul.cpp

```diff
@@ -177,25 +241,61 @@ TEST(TransformationTests, RemoveTransposeBeforeMatmulTestReshapeInOutEq) {
+            }
+        }
+    }
 }
 
 TEST(TransformationTests, RemoveTransposeAfterMatmulTest) {
-    RunTest(
-        handle_transpose_after_matmul::CreateMatmulTransposeFunction({4, 1}, {1, 8}, {2, 16}, false),
-        handle_transpose_after_matmul::CreateMatmulFunction({4, 1}, {1, 8}, {2, 16}, true));
+    for (auto enable_last_reshape : { true, false }) {
+        for (auto enable_add : { true, false }) {
+            for (auto matmul_on_left_side : { true, false }) {
+                for (auto enable_fq : { true, false }) {
+                    RunTest(
+                        handle_transpose_after_matmul::CreateMatmulTransposeFunction(
+                            {4, 1}, {1, 8}, {2, 16}, false, enable_last_reshape, enable_add, matmul_on_left_side, enable_fq),
+                        handle_transpose_after_matmul::CreateMatmulFunction(
+                            {4, 1}, {1, 8}, {2, 16}, true, enable_last_reshape, enable_add, matmul_on_left_side, enable_fq));
+                }
+            }
+        }
+    }
 }
 
 TEST(TransformationTests, RemoveTransposeAfterMatmulTestReshapeInOutEq) {
-    RunTest(
-        handle_transpose_after_matmul::CreateMatmulTransposeFunction({4, 1}, {1, 8}, {8, 4}, false),
-        handle_transpose_after_matmul::CreateMatmulTransposeFunction({4, 1}, {1, 8}, {8, 4}, false));
+    for (auto enable_add : { true, false }) {
+        for (auto matmul_on_left_side : { true, false }) {
+            for (auto enable_fq : { true, false }) {
+                RunTest(
+                    handle_transpose_after_matmul::CreateMatmulTransposeFunction(
+                        {4, 1}, {1, 8}, {8, 4}, false, true, enable_add, matmul_on_left_side, enable_fq),
+                    handle_transpose_after_matmul::CreateMatmulTransposeFunction(
+                        {4, 1}, {1, 8}, {8, 4}, false, true, enable_add, matmul_on_left_side, enable_fq));
+            }
+        }
+    }
 }
 
 TEST(TransformationTests, InsertTransposeAfterMatmulTestReshapeInOutEq) {
-    RunTest(
-        handle_transpose_after_matmul::CreateMatmulFunction({4, 1}, {1, 8}, {4, 8}, false),
-        handle_transpose_after_matmul::CreateMatmulFunction({4, 1}, {1, 8}, {4, 8}, false));
+    for (auto enable_last_reshape : { true, false }) {
+        for (auto enable_add : { true, false }) {
+            for (auto matmul_on_left_side : { true, false }) {
+                for (auto enable_fq : { true, false }) {
+                    RunTest(
+                        handle_transpose_after_matmul::CreateMatmulFunction(
+                            {4, 1}, {1, 8}, {4, 8}, false, enable_last_reshape, enable_add, matmul_on_left_side, enable_fq),
+                        handle_transpose_after_matmul::CreateMatmulFunction(
+                            {4, 1}, {1, 8}, {4, 8}, false, enable_last_reshape, enable_add, matmul_on_left_side, enable_fq));
+                }
+            }
+        }
+    }
```

Comment:
I'd also add/update existed functional tests (functional/plugin/gna/pass_tests/convert_matmul_to_fullyconnected.cpp and functional/plugin/gna/pass_tests/insert_transpose_before_matmul.cpp) to check that all configurations which will take part in these transformations will give us correct final results.

Reply:
done